x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/helpers.ts

```diff
@@ -570,16 +549,16 @@ export const formatDefineStepData = (defineStepData: DefineStepRule): DefineStep
 
 export const formatScheduleStepData = (scheduleData: ScheduleStepRule): ScheduleStepRuleJson => {
   const { ...formatScheduleData } = scheduleData;
-  if (!isEmpty(formatScheduleData.interval) && !isEmpty(formatScheduleData.from)) {
-    const { unit: intervalUnit, value: intervalValue } = getTimeTypeValue(
-      formatScheduleData.interval
-    );
-    const { unit: fromUnit, value: fromValue } = getTimeTypeValue(formatScheduleData.from);
-    const duration = moment.duration(intervalValue, intervalUnit);
-    duration.add(fromValue, fromUnit);
-    formatScheduleData.from = `now-${duration.asSeconds()}s`;
+  const intervalMs = convertTimeDurationToMs(formatScheduleData.interval ?? '');
+  const lookBackMs = convertTimeDurationToMs(formatScheduleData.from ?? '');
+
+  if (intervalMs !== undefined && lookBackMs !== undefined) {
+    const fromSecs = Math.round((intervalMs + lookBackMs) / 1000);
+
+    formatScheduleData.from = `now-${fromSecs}s`;
```

Comment:
This doesn't persist the original value that the user puts in the form.

If I specify `interval = 50m` and `lookback = 6m`, I would expect the rule to be saved as:

```
interval = 50m
from = now-56m
to = now
```

Instead the rule is being saved as:

```
interval = 50m
from = now-3360s
to = now
```

This causes any prebuilt rule to be customized on editing, even if you don't change anything. It's the same effect as https://github.com/elastic/kibana/issues/203315 but a different reason. @dplumlee just FYI.

It looks like the logic was always there, but we need to fix it for Milestone 3.

Reply:
I believe this would be the correct logic to use - the one from the upgrade flyout:

```ts
function serializer(formData: FormData): {
  rule_schedule: RuleSchedule;
} {
  const interval = formData.interval as string;
  const lookback = formData.lookback as string;

  const intervalMs = convertTimeDurationToMs(interval);
  const lookbackMs = convertTimeDurationToMs(lookback);

  invariant(intervalMs !== undefined && intervalMs > 0, 'Rule interval is invalid');
  invariant(lookbackMs !== undefined && lookbackMs >= 0, "Rule's look-back is invalid");

  const from = `now-${toLargestTimeDuration(intervalMs + lookbackMs)}`;

  return {
    rule_schedule: {
      interval: formData.interval,
      from,
      to: 'now',
    },
  };
}
```